futures.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const fs            = require('fs');
+
+const Future        = require('fluture');
+
+const exit0         = require('./common/exit0');
+const exit1         = require('./common/exit1');
+const join          = require('./common/join');
+const S             = require('./common/sanctuary');
+
+
+//    readFile :: String -> Future Error String
+const readFile =
+S.compose(Future.node, S.curry3(fs.readFile, S.__, {encoding: 'utf8'}));
+
+//    concatFiles :: (String -> String) -> Future Error String
+const concatFiles = path =>
+  S.pipe([path,                                   // :: String
+          readFile,                               // :: Future Error String
+          S.map(S.lines),                         // :: Future Error (Array String)
+          S.map(S.map(path)),                     // :: Future Error (Array String)
+          S.chain(S.traverse(Future, readFile)),  // :: Future Error (Array String)
```

Comment:
It saddens me\* to prefer the second version, because it's more concise and it's clearer what's going on. It's also more similar to many of the Promise based solutions in the project.

\* It would be nicer to be able to utilize `traverse` for this purpose without the verbose type juggling. If we would go for the first approach, some ways to make it more readable would be to name `S.compose(Future.Par,  readFile)` something like `concurrentReadFile`, and maybe use `map` to avoid the nested `compose`:

```js
const concurrentReadFile = S.compose(Future.Par, readFile);
//...
  S.map(S.traverse(Future.Par, concurrentReadFile)), // :: Future Error (ConcurrentFuture Error (Array String))
  S.chain(Future.seq)                                // :: Future Error (Array String)
```